### PR TITLE
Problem: no cluster benchmarking

### PR DESCRIPTION
**Pull request by yihuang**
- Integrate testground to do network benchmarking
- Implement python-sdk for testground

### NOTE

Need this patch on testground to support the docker image building with nix: https://github.com/testground/testground/pull/1594

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

## Summary by CodeRabbit

- **New Features**
  - Introduced comprehensive documentation for getting started with Testground.
  - Added functionalities for managing and executing benchmarks, including network settings, synchronization, and peer bootstrapping.
  - Added support for patching TOML and JSON files.
  - Configured Python project dependencies and entry points for benchmarking tasks.

- **Bug Fixes**
  - Updated `.gitignore` to correctly ignore the `result` directory.

- **Documentation**
  - Added a detailed `README.md` for Testground setup and usage.

- **Chores**
  - Updated `CHANGELOG.md` to reflect the integration of Testground for cluster benchmarks.

### CODE REVIEW

**Comment by mmsqe** [comment]
do we need .env.toml for benchmark?

**Comment by yihuang** [comment]
> do we need .env.toml for benchmark?

not for `local:docker` runner

File: testground/README.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+Import the test plan before the first run:
+
+```bash
+$ TESTGROUND_HOME=$PWD/data testground plan import --from /path/to/cronos/testground/benchmark
+```
+
+Run the benchmark test plan locally:
+
+```
+$ testground run single --plan=benchmark --instances=2 --testcase=entrypoint --builder=docker:generic --runner=local:docker --wait
```
Comment (mmsqe) [inline testground/README.md:42]: do we need specify `TESTGROUND_HOME=$PWD/data` here

Comment (yihuang) [reply testground/README.md:42]: No, it just communicate with the daemon

Comment (mmsqe) [reply testground/README.md:42]: for local:docker, do we need add Dockerfile like in [example](https://github.com/testground/testground/blob/master/plans/example-browser-node), I need run `TESTGROUND_HOME=$PWD/data testground run single --plan=example-browser-node --instances=2 --testcase=sync --builder=docker:generic --runner=local:docker --wait`

Comment (yihuang) [reply testground/README.md:42]: we need to use `docker:nix` builder, supported by the patched testground linked in the PR description.

**Review by mmsqe — APPROVED** [review]

**Pull request merged**